soc/Kconfig

```diff
@@ -11,3 +11,24 @@ osource "$(SOC_DIR)/$(ARCH)/Kconfig"
 osource "$(SOC_DIR)/$(ARCH)/*/Kconfig"
 
 endmenu
+
+#
+# SOC_*_LD: SoC specific Linker script additions
```

Comment:
There is no reasonable way for a non-ARM user to discover that he can't use this feature. We either need to make the Kconfig options depend on ARM or add support for all platforms.

Reply:
Added the #include for common linker scripts under include/, and added conditionals for the Kconfig.